src/pages/Signup/Signup.js

```diff
@@ -0,0 +1,118 @@
+import { Box, Button, Flex, Heading, Input, Text } from "@chakra-ui/react";
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { Link, useNavigate } from "react-router-dom";
+import { signupUser } from "../../features/authenticationSlice";
+
+const Signup = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [userCredentials, setUserCredentials] = useState({
+    username: "",
+    password: "",
+    firstName: "",
+    lastName: "",
+  });
+
+  const signupHandler = () => {
+    if (
+      userCredentials.username !== "" &&
+      userCredentials.password !== "" &&
+      userCredentials.firstName !== "" &&
+      userCredentials.lastName !== ""
+    ) {
```

Comment:
It's Ok for now, but later on you can provide corresponding reason for signUp handler for not proceeding with the signup(In case the user credentials are empty or invalid).